app/models/smart_multilingual_tokenizer.rb

```diff
@@ -0,0 +1,35 @@
+class SmartMultilingualTokenizer
+  Response = Data.define(:token, :start_offset, :end_offset, :type)
+  Analyzed = Data.define(:language, :tokens)
+  def initialize
+    @client = Elasticsearch::Client.new(hosts: [ "localhost:9200" ])
+    @index_name = "smart_multilingual"
+  end
+
+  # Analyze text: detect language and tokenize using Elasticsearch analyzer
+  def analyze_multilingual_text(text)
+    Analyzed.new(detect_language(text),
+                 tokenize_with_standard_analyzer(text))
+  end
+
+  private
+
+  # Detect language using CLD3 (refactored to be more idiomatic Ruby)
+  def detect_language(text)
+    lang = CLD3::NNetLanguageIdentifier.new(0, 400).find_language(text)&.language
+    { ja: "ja", ko: "ko", en: "en" }[lang] || "unknown"
```

Comment:
あまり直感的に読めないコードだと思いました。
次のような代替案はどうでしょうか？

```suggestion
    %w[ja ko en].include?(lang) ? lang : "unknown"
```

あるいは

```suggestion
  case lang
  when "ja", "ko", "en"
    lang
  else
    "unknown"
  end
```

定数SUPPORTED_LANGSを定義して

```suggestion
    SUPPORTED_LANGS.include?(lang) ? lang : "unknown"
```

とすると意図が分かりやすいコードになるかもしれません。